qiskit/compiler/scheduler.py

```diff
@@ -34,7 +36,11 @@ def _log_schedule_time(start_time, end_time):
     log_msg = "Total Scheduling Time - %.5f (ms)" % ((end_time - start_time) * 1000)
     logger.info(log_msg)
 
-
+@deprecate_arg(
+    "backend",
+    additional_msg=("'backend' argument becomes required from Optional."),
+    since="0.25.0",
+)
```

Comment:
Target is made by some backend parameters and I would like to make backend required from Optional.

Reply:
@to24toro 
I think this is an incorrect use of `@deprecate_arg`. The warning will only be raised when the user *does* provide the argument explicitly. This is the opposite of what you want - you want the warning to be raised when the user doesn't provide it.
I would have placed the warning in the function itself:
```
if backend is None:
  # raise warning
```